pkg/controller/datavolume/util.go

```diff
@@ -77,8 +76,10 @@ func pvcFromStorage(client client.Client, recorder record.EventRecorder, log log
 	if err != nil {
 		return nil, err
 	}
-
 	if storageClass == nil {
+		if err := updatePvcSpecFromPvcOrPv(client, pvcSpec, pvc); err != nil {
```

Comment:
I think this flow is getting confusing, it'd be good to have some comments explaining why are we doing this. I'm also not a fan of the `updatePvcSpecFromPvcOrPv` name, it's not a must but I think this whole storageClass handling would benefit from being in a separate function.

Reply:
Simplified and renamed.